[fix] Improve node label selection and dynamic label visibility

cestercian opened the pull request:
**Node labels now prefer `label`, then `name`, then `id` for display. Added dynamic label visibility in graph mode based on zoom scale, controlled by `graphLabelScaleThreshold` and scale limits, with debounce to avoid excessive updates.**

## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).  
- [x] I have manually tested the changes proposed in this pull request.  
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.  
- [x] I have updated the documentation.  

## Reference to Existing Issue

Closes #148.

## Description of Changes

- Improved node label selection logic:  
  - Prefer `node.label` → fallback to `node.name` → fallback to `node.id`.  
- Introduced **dynamic label visibility** in graph mode:  
  - Controlled by `graphLabelScaleThreshold` (default: `1.0`).  
  - Respects scale limits defined in config.  
  - Uses debounce to prevent excessive updates.  
  - Ensures zoom/roam transform is not reset.  
- Added resize handling to maintain responsiveness.  

## Screenshot
<img width="1734" height="1114" alt="image" src="https://github.com/user-attachments/assets/d5e6ab33-c4a3-4ee2-9faa-ab0790c4f2b6" />


<img width="1734" height="1114" alt="image" src="https://github.com/user-attachments/assets/f5d21b80-548d-4e45-bdbc-4970c80c69aa" />

nemesifier reviewed (commented):
Good progress on this @cestercian, it looks encouraging!

Here's a few details I have observed.

## Labels are visible on initial page load with default settings

When I load the basic example, the labels are visible:

![labels](https://github.com/user-attachments/assets/0836c115-f312-42db-94e3-de3503618522)

Consider a situation like the following, this is a screenshot from a prod system taken right after page load:

<img width="1920" height="931" alt="Screenshot from 2025-08-21 15-37-56" src="https://github.com/user-attachments/assets/8d33116d-0fbc-4063-948c-7fedc1c2fc80" />

On real world systems, when the chart is busy, it doesn't make sense to show the labels after the page loads, it would be better to show the only after a certain level of zoom.

Can we configure the default setting of the library so that this happens automatically?

The users will still be able to tweak the settings if needed.

I expanded these concepts in https://github.com/openwisp/netjsongraph.js/issues/420, which we can work on at a later stage.

cestercian commented on src/js/netjsongraph.render.js:
```diff
@@ -365,6 +374,64 @@ class NetJSONGraphRender {
+    // Graph mode: use dedicated graphLabelScaleThreshold (scale factor), default 1.0
+    const threshold =
+      typeof self.config.graphLabelScaleThreshold === "number"
+        ? self.config.graphLabelScaleThreshold
```
addressed in commit https://github.com/openwisp/netjsongraph.js/pull/419/commits/60ed602cb00d9fe0f8cd1fe590d32dce8cf8db42

cestercian commented:
https://github.com/user-attachments/assets/c30616b4-640f-44e9-bd77-23cf89511c01



Hi @nemesifier — I’ve addressed your changes and fixed a related bug.

## Graph Mode
- Added `showGraphLabelsAtZoom` to gate labels by series zoom without resets.
- Switched label visibility to a formatter; no more `setOption`-driven snap-backs.
- Introduced stable series IDs (`graph-series`) and use series-only updates.

## Labels
- Fallback to `node.name` or `String(node.id)` when `node.label` is missing.

nemesifier reviewed (changes requested):
@cestercian let's proceed with the improvements discussed today, see my comments below for further input.

nemesifier reviewed (changes requested):
After the latest changes the graph moves all the time, not just when the threshold is crossed.

PS: false alarm.
